### DISCUSSION TITLE
jooq-checker PlainSQLChecker

### DISCUSSION THREAD

**Discussion by nicchongwb** [jOOQ/jOOQ#14647] [Q&A] [answered]
Hi has anyone managed to use jooq-checker PlainSQLChecker with checkerframework for a gradle Kotlin project?

Also I will like to understand if PlainSQLChecker implementation only works on parsing Java syntax or it is able to handle Kotlin too.

**Comment by lukaseder** [comment, marked as answer]
Thanks for your message. We currently do not support this scenario. Both implementations of the checker (using error prone or the checker framework) rely on javac specific API to do their processing, so it's unlikely that it works for kotlin out of the box.

**Reply by lukaseder** [reply]
In case a follow-up question would be if we plan on supporting this: The answer is no as well, currently. The maintenance of the checker is hard enough for Java, given that:

- neither checkerframework nor error prone are maintained as "full fledged products" with multi JDK support
- both rely on internal javac APIs heavily

It will be even harder to maintain a checker also for kotlin at the same time.
